gufe/chemicalsystem.py

```diff
@@ -12,7 +12,7 @@ class ChemicalSystem(GufeTokenizable, abc.Mapping):
     def __init__(
         self,
         components: dict[str, Component],
-        name: Optional[str] = "",
+        name: str | None = "",
```

Comment:
So I get that this is the new syntax, but `Optional` is, in my opinion, way more descriptive than `a | None`. 

`Optional[a]` > `a | None` > `Union[a, None]`

Reply:
It does save having to import `Optional` -- I think Optional is more descriptive, but ` | None` is more explicit. Since `Optional` means "can be `None`" in this context, we are now exhaustively listing all the types it can be. I think in a doc string "Optional" is way more descriptive (I would never say "can be None" in a doc string, I would say the argument is optional) but in the code, I like the explicitness.

Reply:
:( I had finally given up on my love for `Union`